android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenDeviceIconManager.java

```diff
@@ -73,13 +105,13 @@ void saveFileToCache(Bitmap icon, String iconUrl) {
         writeDeviceIconParametersToSharedPreferences(iconHash);
     }
 
-    Bitmap getFileFromCache(String iconUrl) {
+    private Bitmap getFileFromCache(String iconUrl) {
         String iconHash = getMD5HashFromIconUrl(iconUrl);
         SharedPreferences sharedPref = this.context.getSharedPreferences(SDL_DEVICE_STATUS_SHARED_PREFS, Context.MODE_PRIVATE);
         String iconLastUpdatedTime = sharedPref.getString(iconHash, null);
 
         if (iconLastUpdatedTime != null) {
-            Bitmap cachedIcon = BitmapFactory.decodeFile(this.context.getCacheDir() + "/" + STORED_ICON_PATH + "/" + iconHash);
+            Bitmap cachedIcon = BitmapFactory.decodeFile(this.context.getCacheDir() + "/" + STORED_ICON_DIRECTORY_PATH + "/" + iconHash);
             if(cachedIcon == null) {
                 DebugTool.logError("Failed to get Bitmap from decoding file cache");
                 clearIconDirectory();
```

Comment:
I don't think that this is necessarily what we want to do here, unlike iOS. On iOS, all the image file information is contained in one file. So on iOS, when you clear the icon directory, the file went along with it. Here, if the directory is cleared, all the icon information remains in `SharedPreferences`. That means that attempting to retrieve an icon may fail and then we'll have a cascading set of retrieval failures because old data is contained in `SharedPreferences` that we know isn't accurate.

I can think of a few possibilities to solve this:
1. We don't clear the icon directory if an error occurs. This isn't aligned with iOS' mitigation strategy and has the possibility of leaving icons in the directory that are unused.
2. Instead of storing the data in separate SharedPreferences keys, use a single key, store the information in JSON and store the JSON in the SharedPreferences under that single key. Then when you clear the icon directory, you clear the JSON document as well.

Reply:
I see what you mean, we would also have the option of sharedPref.edit().clear().commit();

https://developer.android.com/reference/android/content/SharedPreferences.Editor#clear()

would this be closer to the ios implementation?

Reply:
If that clears all data about cached icons and their expiration dates, then yes, I believe it would.